examples/plugins/ebpf/main.go

```diff
@@ -0,0 +1,179 @@
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"net/http"
+	"os"
+	"os/exec"
+	"sync"
+	"time"
+)
+
+func main() {
+	hostname, _ := os.Hostname()
+	var (
+		addr   = flag.String("addr", "/var/run/scope/plugins/ebpf.sock", "unix socket to listen for connections on")
+		hostID = flag.String("hostname", hostname, "hostname of the host running this plugin")
+	)
+	flag.Parse()
+
+	log.Println("Starting...")
+
+	// Compile and run the ebpf script
+	ebpf := exec.Command("http-parse-simple.py")
+	ebpf.Stderr = os.Stderr
+	stdout, err := ebpf.StdoutPipe()
+	if err != nil {
+		log.Fatal(err)
+	}
+	if err := ebpf.Start(); err != nil {
+		log.Fatal(err)
+	}
+	defer func() {
+		if ebpf.Process != nil {
+			ebpf.Process.Kill()
+		}
+	}()
+
+	os.Remove(*addr)
+	listener, err := net.Listen("unix", *addr)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer func() {
+		listener.Close()
+		os.Remove(*addr)
+	}()
+
+	log.Printf("Listening on: unix://%s", *addr)
+
+	plugin := &Plugin{HostID: *hostID}
+	go plugin.loop(stdout)
+	http.HandleFunc("/", plugin.Handshake)
+	http.HandleFunc("/report", plugin.Report)
+	if err := http.Serve(listener, nil); err != nil {
+		log.Printf("error: %v", err)
+	}
+	ebpf.Wait()
+}
+
+type Plugin struct {
+	sync.Mutex
+	HostID string
+	tuples []tuple
+}
+
+func (p *Plugin) Handshake(w http.ResponseWriter, r *http.Request) {
+	log.Printf("Probe %s handshake", r.FormValue("probe_id"))
+	err := json.NewEncoder(w).Encode(map[string]interface{}{
+		"name":        "ebpf",
+		"description": "Adds a graph of http requests/second to processes",
```

Comment:
From the UI perspective, it adds this graph to processes. It adds the metrics onto the endpoints, which get rolled up into processes in the app, but it's not strictly true if a process has > 1 listening addr as the metrics won't be summed. Am working on this bit now, to see if there's a better way.